File: code/python/tests-workshop/cdk_workshop/cdk_workshop_stack.py
from constructs import Construct
from aws_cdk import (
    core as cdk,
    aws_lambda as _lambda,
    aws_apigateway as apigw,
)
from cdk_dynamo_table_viewer import TableViewer
from cdk_workshop.hitcounter import HitCounter


class CdkWorkshopStack(cdk.Stack):
    def __init__(self, scope: Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Defines an AWS Lambda resource
        hello = _lambda.Function(
            self,
            "HelloHandler",
            runtime=_lambda.Runtime.PYTHON_3_7,
            code=_lambda.Code.from_asset("lambda"),
            handler="hello.handler",
        )

        hello_with_counter = HitCounter(self, "HelloHitCounter", downstream=hello)

        apigw.LambdaRestApi(self, "Endpoint", handler=hello_with_counter._handler)

        TableViewer(
            self, "ViewHitCounter", title="Hello Hits", table=hello_with_counter.table
        )

File: code/csharp/test-workshop/src/CdkWorkshop/CdkWorkshopStack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.APIGateway;
using Amazon.CDK.AWS.Lambda;
using Constructs;
using Cdklabs.DynamoTableViewer;
namespace CdkWorkshop
{
    public class CdkWorkshopStack : Stack
    {
        // Defines a new lambda resource
        public CdkWorkshopStack(Construct scope, string id) : base(scope, id)
        {
            var hello = new Function(this, "HelloHandler", new FunctionProps
            {
                Runtime = Runtime.NODEJS_14_X,
                Code = Code.FromAsset("lambda"),
                Handler = "hello.handler"
            });

            var helloWithCounter = new HitCounter(this, "HelloHitCounter", new HitCounterProps
            {
                Downstream = hello
            });

            new LambdaRestApi(this, "Endpoint", new LambdaRestApiProps
            {
                Handler = helloWithCounter.Handler
            });

            new TableViewer(this, "ViewerHitCount", new TableViewerProps
            {
                Title = "Hello Hits",
                Table = helloWithCounter.MyTable
            });
        }
    }
}
